overlord/devicestate/handlers.go

```diff
@@ -361,12 +435,29 @@ func prepareSerialRequest(t *state.Task, privKey asserts.PrivateKey, device *aut
 		headers["serial"] = cfg.proposedSerial
 	}
 
+	for k, v := range regCtx.SerialRequestExtraHeaders() {
+		headers[k] = v
+	}
+
 	serialReq, err := asserts.SignWithoutAuthority(asserts.SerialRequestType, headers, cfg.body, privKey)
 	if err != nil {
 		return "", err
 	}
 
-	return string(asserts.Encode(serialReq)), nil
+	buf := new(bytes.Buffer)
```

Comment:
We don't use `new()` much, how about `buf := &bytes.Buffer{}` instead?

Reply:
That's a bit of an odd request, we do use a bit of both, I didn't make go support both.